integrations/operator/sidecar/tbot.go

```diff
@@ -139,6 +138,42 @@ func (b *Bot) GetClient(ctx context.Context) (*client.Client, error) {
 	return c, trace.Wrap(err)
 }
 
+// GetClient gets a client authenticated as the bot. To avoid rebuilding a client for each call, this function caches
+// the client and creates a new one only when the client certs changed (tbot renewed them).
+func (b *Bot) GetClient(ctx context.Context) (*client.Client, error) {
+	b.mutex.Lock()
+	defer b.mutex.Unlock()
+
+	if !b.running {
+		return nil, trace.Errorf("bot not started yet")
+	}
+	// If the bot has not joined the cluster yet or not generated client certs we bail out
+	// This is either temporary or the bot is dead and the manager will shut down everything.
+	storageDestination := b.cfg.Storage.Destination
+	if botCert, err := storageDestination.Read(ctx, identity.TLSCertKey); err != nil || len(botCert) == 0 {
+		return nil, trace.Retry(err, "bot cert not yet present")
+	}
+
+	cert, err := b.cfg.Outputs[0].GetDestination().Read(ctx, identity.TLSCertKey)
```

Comment:
When copying this line I realized this was likely subject to races as the MemoryDestination doesn't lock when reading/writing. (tbot could attempt to write to the store while we're reading) I did not change this in the PR but can add a mutex to protect the `MemoryDestination.store` if @strideynet agrees.

Reply:
A RWMutex on `DestinationMemory` is a good idea - the memory destination wasn't really originally meant to be used and hence is missing one. Feel free to put one in in this PR - it should only be a few lines change :')

Reply:
Fixed in ad6b722d371226bb1f07bc42e6ad81a454ec4f90